sdk/keyvault/keyvault-certificates/review/keyvault-certificates.api.md

```diff
@@ -260,32 +260,34 @@ export interface ErrorModel {
     readonly message?: string;
 }
 
+export { ExtendedCommonClientOptions }
```

Comment:
why is it re-exported?

Reply:
I'm not sure, the only place where this is being exported is in `index.ts` on [line 197](https://github.com/JonathanCrd/azure-sdk-for-js/blob/4014b39a3c483489c3925550aebef50f0ae1436a/sdk/keyvault/keyvault-certificates/src/index.ts#L197).

`CommonClientOptions`  from Core-http was also being exported there, I just renamed it to the equivalent in CoreV2.

Reply:
I this change is breaking then but I wonder what was the point of re-exporting the old interface in the first place. I am thinking of biting the bullet and removing the re-export even though that is also a breaking change. @xirzec what do you think?

Reply:
For the record, `PipelineOptions` from core-http (which has been removed in this PR) was being exported too in the same way as `ExtendedCommonClientOptions`. 
You can find it here: 
https://github.com/JonathanCrd/azure-sdk-for-js/blob/main/sdk/keyvault/keyvault-certificates/review/keyvault-certificates.api.md?plain=1#L433

I'm also curious why that's the behavior, however, wouldn't it be a breaking change anyway since `PipelineOptions` is no longer being exported?

Reply:
I would like to get this PR merged in the coming days, but want to resolve this first. I did a bit of digging to try and understand the impact of the change. `PipelineOptions` was initially exported in [this commit from 3 years back](https://github.com/Azure/azure-sdk-for-js/commit/2f973f57f8c1916eaada2f399c52fbb62de54cc0#diff-b5a2a450cdea87e9ea8fb2dd2b7a1375673c20e26a61460f9c371ad11d3d2e88R237) and hasn't been touched since. At the time of that commit, the [CertificateClient constructor](https://github.com/Azure/azure-sdk-for-js/commit/2f973f57f8c1916eaada2f399c52fbb62de54cc0#diff-b5a2a450cdea87e9ea8fb2dd2b7a1375673c20e26a61460f9c371ad11d3d2e88R16) took `PipelineOptions` as a parameter, so you can see why the export was required then. This was extracted out to a `CertificateClientOptions` by Daniel [here](https://github.com/Azure/azure-sdk-for-js/commit/776983421e2#diff-b5a2a450cdea87e9ea8fb2dd2b7a1375673c20e26a61460f9c371ad11d3d2e88R51), which is what we have now.

I don't see the point in exporting the new `ExtendedCommonClientOptions` as is here -- let's remove that at least. If we were to remove the export of `PipelineOptions`, would we still be able to release this as a minor? I can think of some hackery that might be less breaking (re-exporting `ExtendedCommonClientOptions` as `PipelineOptions`...) but that doesn't sit very well with me since we're effectively giving the type an incorrect name.

Thoughts, @deyaaeldeen, @xirzec, @joheredi (and anybody else)?

Reply:
To wrap this up: After an offline discussion, we're in agreement it would be ok to remove the export of `PipelineOptions` and still release this as a minor.

Cc @deyaaeldeen @joheredi @bterlson